src/runtime/arc.ts

```diff
@@ -738,7 +738,10 @@ ${this.activeRecipe.toString()}`;
     this.storeTags.set(store, new Set(tags));
 
     this.storageKeys[store.id] = store.storageKey;
-    store.on(() => this._onDataChange());
+    store.on(async () => {
```

Comment:
Might be simpler to have onDataChange return true? (I've seen a couple of these).

Reply:
Done